mypy_django_plugin/lib/helpers.py

```diff
@@ -385,3 +386,36 @@ def add_new_manager_base(api: SemanticAnalyzerPluginInterface, fullname: str) ->
     if sym is not None and isinstance(sym.node, TypeInfo):
         bases = get_django_metadata_bases(sym.node, "manager_bases")
         bases[fullname] = 1
+
+
+def resolve_lazy_reference(
+    reference: str, *, api: Union[TypeChecker, SemanticAnalyzer], django_context: "DjangoContext", ctx: Context
+) -> Optional[TypeInfo]:
+    """
+    Attempts to resolve a lazy reference(e.g. "<app_label>.<object_name>") to a
+    'TypeInfo' instance.
+    """
+    if "." not in reference:
+        # <object_name> -- needs prefix of <app_label>. We can't implicitly solve
+        # what app label this should be, yet.
+        api.fail("Could not resolve lazy reference without an app label", ctx)
+        api.note(
+            ("Try to use a reference explicitly prefixed with app label:" f' "<app_label>.{reference}" instead'),
+            ctx,
+        )
+        return None
+
+    # Reference conforms to the structure of a lazy reference: '<app_label>.<object_name>'
+    fullname = django_context.model_class_fullnames_by_label_lower.get(reference.lower())
```

Comment:
Question about `lower`: why is that needed?

Reply:
The implementation is case insensitive:

> Returns the [Model](https://docs.djangoproject.com/en/4.2/ref/models/instances/#django.db.models.Model) with the given __app_label__ and __model_name__. As a shortcut, this method also accepts a single argument in the form __app_label.model_name__. __model_name__ is case-insensitive.

Ref: https://docs.djangoproject.com/en/4.2/ref/applications/#django.apps.apps.get_model